### PR TITLE
[DOCS] Clarify CCR auto-follow requirements for data streams

### PR DESCRIPTION
**Pull request by danhermann**
When configuring an auto-follow pattern that includes data streams, the `follow_index_pattern` option must not change the index name as differing backing index names on leader and follower data streams are not supported.

### CODE REVIEW

**Review by martijnvg — APPROVED** [review]
👍, I think we can only support this if the data stream name is renamed too.

File: docs/reference/ccr/apis/auto-follow/put-auto-follow-pattern.asciidoc [hunk 1]
```diff
@@ -75,7 +75,9 @@ indices.
 `follow_index_pattern`::
   (Optional, string) The name of follower index. The template `{{leader_index}}`
   can be used to derive the name of the follower index from the name of the
-  leader index.
+  leader index. When following a data stream, `{{leader_index}}` should be used
+  as changing the names of a follower data stream's backing indices is not
+  supported.
```
Comment (jrodewig) [inline docs/reference/ccr/apis/auto-follow/put-auto-follow-pattern.asciidoc:80]: Minor reword to break this into two sentences and avoid passive voice.
```suggestion
  leader index. When following a data stream, use `{{leader_index}}`; {ccr-init}
  does not support changes to the names of a follower data stream's backing
  indices.
  ```

**Review by jrodewig — APPROVED** [review]
LGTM. I left a non-blocking suggestion, but feel free to ignore if wanted. Thanks @danhermann.

**Comment by danhermann** [comment]
Thanks, @jrodewig and @martijnvg!

**Pull request merged**